libs/cosmos-sdk/baseapp/baseapp.go

```diff
@@ -1006,3 +1007,29 @@ func (app *BaseApp) GetRawTxInfo(rawTx tmtypes.Tx) mempool.ExTxInfo {
 
 	return app.GetTxInfo(app.checkState.ctx, tx)
 }
+
+func (app *BaseApp) GetTxHistoryGasUsed(rawTx tmtypes.Tx) int64 {
+	tx, err := app.txDecoder(rawTx)
+	if err != nil {
+		return -1
+	}
+
+	txFnSig, toDeployContractSize := tx.GetTxFnSignatureInfo()
+	if txFnSig == nil {
+		return -1
+	}
+
+	db := InstanceOfGasUsedRecordDB()
+	data, err := db.Get(txFnSig)
+	if err != nil || len(data) == 0 {
+		return -1
+	}
+
+	if toDeployContractSize > 0 {
+		// if deploy contract case, the history gas used value is unit gas used
+		return int64(binary.BigEndian.Uint64(data)) * int64(toDeployContractSize)
```

Comment:
it's better to add `10000` gas to cover the option of account change or any other operations of gas consumed